test/black-box/itemimage/ListingItemTemplateFeatureImageCommand.test.ts

```diff
@@ -0,0 +1,220 @@
+
+describe('ListingItemTemplateFeatureImageCommand', () => {
+
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = process.env.JASMINE_TIMEOUT;
+
+    const log: LoggerType = new LoggerType(__filename);
+    const testUtil = new BlackBoxTestUtil();
+
+    const itemImageCommand = Commands.ITEMIMAGE_ROOT.commandName;
+    const featuredImgaeCommand = Commands.ITEMIMAGE_FEATURED.commandName;
+
+
+    let createdItemImageIdNew;
+    let listingItemId;
+    let listingItems;
+
+    const keys = [
+        'id', 'hash', 'updatedAt', 'createdAt'
+    ];
+
+    const testDataListingItemTemplate = {
+        profile_id: 0,
+        hash: '',
+        itemInformation: {
+            title: 'item title1',
+            shortDescription: 'item short desc1',
+            longDescription: 'item long desc1',
+                itemCategory: {
+                    key: 'cat_high_luxyry_items'
+                }
+        },
+        paymentInformation: {
+            type: PaymentType.SALE
+        }
+    } as ListingItemTemplateCreateRequest;
+
+    const noListingTemplateID = [];
+    const noFeaturedImage = [1];
+    const WrongTypeListingTemplateID = ['string', 1];
+    const wrongTypenoFeaturedImage = [1, 'string'];
+
+    let defaultProfile: resources.Profile;
+
+    const generateListingItemTemplateParams = new GenerateListingItemTemplateParams([
+        true,   // generateItemInformation
+        true,   // generateItemLocation
+        true,   // generateShippingDestinations
+        false,   // generateItemImages
+        true,   // generatePaymentInformation
+        true,   // generateEscrow
+        true,   // generateItemPrice
+        true,   // generateMessagingInformation
+        false    // generateListingItemObjects
+    ]).toParamsArray();
+
+
+
+    beforeAll(async () => {
+        await testUtil.cleanDb();
+
+        // get default profile and market
+        defaultProfile = await testUtil.getDefaultProfile();
+
+        const generateListingItemParams = new GenerateListingItemParams([
+            true,   // generateItemInformation
+            true,   // generateItemLocation
+            false,   // generateShippingDestinations
+            false,   // generateItemImages
+            false,   // generatePaymentInformation
+            false,   // generateEscrow
+            false,   // generateItemPrice
+            false,   // generateMessagingInformation
+            false    // generateListingItemObjects
+        ]).toParamsArray();
+
+        defaultProfile = await testUtil.getDefaultProfile();
+        testDataListingItemTemplate.profile_id = defaultProfile.id;
+
+        // set hash
+        testDataListingItemTemplate.hash = ObjectHash.getHash(testDataListingItemTemplate, HashableObjectType.LISTINGITEMTEMPLATE);
+
+        // create item template
+        const addListingItemTempRes: any = await testUtil.addData(CreatableModel.LISTINGITEMTEMPLATE, testDataListingItemTemplate);
```

Comment:
is there a reason to use addData rather than generateData?

Reply:
TBH i dont know, thats what i tried from one of the other black-box tests and it happened to work, not too clued up on the tests of the market